app/controllers/application_controller.rb

```diff
@@ -1,14 +1,24 @@
 class ApplicationController < ActionController::API
   before_action :authenticate_request
-  # TODO: Why do you include an attr_reader here?
   attr_reader :current_user
 
   include ExceptionHandler
 
   private
 
   def authenticate_request
-    @current_user = AuthorizeApiRequest.new(request.headers).call
-    render json: { error: 'Not Authorized' }, status: 401 unless @current_user
+    if request.headers['Authorization'].present?
+      @current_user = AuthorizeApiRequest.new(request.headers).call
+      render status: :forbidden unless @current_user
+    else
+      render status: :unauthorized
+    end
+  end
+
+  def validate_user
+    @chatroom = Chatroom.find(chatroom_params[:id])
```

Comment:
It would be a little bit nicer to look for the chatroom directly on the user model.
The command would be '@current_user.chatrooms.find(chatroom_params[:id])', and it would trigger a 404 error when not available.
The reason for this is that a possible attacker might just try to use random IDs to find out which chatrooms are actually available and which not, resulting in 403 answers for existing chatrooms and 404 for non existing ones.